<launch>
    <arg name="show_output" default="false" />
    <arg name="launch_calibration" default="false" />
    <arg name="aruco_tracker" default="false" />
    <arg name="camera_name" default="logitech_webcam" />
    <arg name="camera_device" default="/dev/video0 " />

    <!-- Start webcam -->
    <node name="$(arg camera_name)" pkg="usb_cam" type="usb_cam_node" output="screen">
        <param name="video_device" value="$(arg camera_device)" />
<<<<<<< HEAD
        <param name="image_width" value="1920" />
        <param name="image_height" value="1080" />
=======
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
>>>>>>> 9e3fc93 (yes)
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="tracking_cam" />
        <param name="camera_name" value="$(arg camera_name)" />
        <param name="camera_info_url" value="package://lab_2/calibration/camera/$(arg camera_name).yaml" />
        <param name="io_method" value="mmap" />
    </node>

    <!-- Rectified image -->
    <node name="rectify_image" pkg="image_proc" type="image_proc" output="screen" ns="$(arg camera_name)" />

    <!-- Start aruco tracker -->
    <group if="$(arg aruco_tracker)">
        <node name="aruco_detect" pkg="aruco_detect" type="aruco_detect" output="screen">
            <param name="fiducial_len" value="0.05" />
            <param name="dictionary" value="5" />
            <param name="publish_images" value="true" />
            <param name="publish_fiducial_tf" value="false" />
            <remap from="camera" to="$(arg camera_name)/image_rect_color" />
            <remap from="camera_info" to="$(arg camera_name)/camera_info" />
            <remap from="/fiducial_images" to="$(arg camera_name)/fiducial_images" />
            <remap from="/fiducial_transforms" to="$(arg camera_name)/fiducial_transforms" />
            <remap from="/fiducial_vertices" to="$(arg camera_name)/fiducial_vertices" />
        </node>
    </group>

    <!-- Show outputs -->
    <group if="$(arg show_output)">
        <!-- Show raw image -->
        <node name="raw_output" pkg="image_view" type="image_view" respawn="true" output="screen">
            <param name="autosize" value="false" />
            <remap from="image" to="$(arg camera_name)/image_raw" />
        </node>
        <!-- Show marker results -->
        <node name="marker_results" pkg="image_view" type="image_view" respawn="true" output="screen">
            <param name="autosize" value="false" />
            <remap from="image" to="$(arg camera_name)/fiducial_images" />
        </node>
    </group>

    <!-- Launch Aruco TF publisher node -->
    <!-- <node name="aruco_tf_node" pkg="shadowcalib" type="aruco_tf" respawn="true" output="screen" /> -->

</launch>